enclave/src/test_proxy.rs

```diff
@@ -0,0 +1,128 @@
+    hashmap.insert(main_account_one, vec![dummy_account_one]);
+    hashmap.insert(main_account_two, vec![dummy_account_one, dummy_account_two]);
+    hashmap.insert(
+        main_account_three,
+        vec![dummy_account_one, dummy_account_two, dummy_account_three],
+    );
+    hashmap
+}
+
+pub fn pointer_initialize() -> sgx_status_t {
+    let mut hashmap: HashMap<[u8; 32], Vec<[u8; 32]>> = get_dummy_map();
+    let polkadex_hashmap: PolkadexAccountsStorage = PolkadexAccountsStorage::from_hashmap(hashmap);
+    let storage_ptr = Arc::new(SgxMutex::<PolkadexAccountsStorage>::new(polkadex_hashmap));
+    let ptr = Arc::into_raw(storage_ptr);
+    GLOBAL_ACCOUNTS_STORAGE.store(ptr as *mut (), Ordering::SeqCst);
+    sgx_status_t::SGX_SUCCESS
+}
+
+pub fn load_proxy_registry() -> SgxResult<&'static SgxMutex<PolkadexAccountsStorage>> {
+    let ptr =
+        GLOBAL_ACCOUNTS_STORAGE.load(Ordering::SeqCst) as *mut SgxMutex<PolkadexAccountsStorage>;
+    if ptr.is_null() {
+        return Err(sgx_status_t::SGX_ERROR_UNEXPECTED);
+    } else {
+        Ok(unsafe { &*ptr })
+    }
+}
+
+#[allow(unused)]
+pub fn test_check_main_account() {
+    pointer_initialize();
+    let account_to_find_real: [u8; 32] = Vec::from("first_account").using_encoded(blake2_256);
+    let account_to_find_false: [u8; 32] = Vec::from("false_account").using_encoded(blake2_256);
+    assert_eq!(polkadex::check_main_account(account_to_find_real), Ok(true));
+    assert_eq!(
+        polkadex::check_main_account(account_to_find_false),
+        Ok(false)
+    );
+}
+
+#[allow(unused)]
+pub fn test_check_proxy_account() {
+    pointer_initialize();
+    let main_account: [u8; 32] = Vec::from("first_account").using_encoded(blake2_256);
+    let main_account_false: [u8; 32] = Vec::from("false_account").using_encoded(blake2_256);
+    let dummy_account_one: [u8; 32] = Vec::from("first_dummy_account").using_encoded(blake2_256);
+    let dummy_account_false: [u8; 32] = Vec::from("false_dummy_account").using_encoded(blake2_256);
+    assert_eq!(
+        polkadex::check_proxy_account(main_account, dummy_account_one),
+        Ok(true)
+    );
+    assert_eq!(
+        polkadex::check_proxy_account(main_account, dummy_account_false),
+        Ok(false)
+    );
+    assert_eq!(
+        polkadex::check_proxy_account(main_account_false, dummy_account_false),
+        Ok(false)
+    );
+}
+
+#[allow(unsued)]
+pub fn test_add_main_account() {
+    pointer_initialize();
+    let main_account: [u8; 32] = Vec::from("new_account").using_encoded(blake2_256);
+    polkadex::add_main_account(main_account);
+    let mutex = load_proxy_registry().unwrap();
+    let mut proxy_storage = mutex.lock().unwrap();
+    assert_eq!(proxy_storage.accounts.contains_key(&main_account), true);
+}
+
+#[allow(unsued)]
+pub fn test_remove_main_account() {
+    pointer_initialize();
+    let main_account: [u8; 32] = Vec::from("first_account").using_encoded(blake2_256);
```

Comment:
Could you ensure here that first_account acutally existed before removal? Otherwise a small change in the pointer_initialize() function might render this test useless without anyone realizing it.

Reply:
resolved